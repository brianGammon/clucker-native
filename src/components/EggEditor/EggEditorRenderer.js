/* @flow */
import React from 'react';
import {
  Button,
  Container,
  Content,
  View,
  Text,
  Header,
  Left,
  Body,
  Right,
  Form,
  CheckBox,
  Picker,
  Label,
  Item,
  Icon,
  Title,
} from 'native-base';
import moment from 'moment';
import { Field, FieldGroup, FieldControl } from 'react-reactive-form';
import DatePicker from '../DatePicker';
import { type Navigation, type Chicken } from '../../types';
import FormInput from '../FormInput';
import styles from './styles';

type Props = {
  mode: 'Edit' | 'Add',
  navigation: Navigation,
  form: any,
  chickens: {
    [string]: Chicken,
  },
  handlePickItem: (itemValue: string) => void,
  toggleDamaged: (damaged: boolean) => void,
  onSaveForm: () => void,
  onDateChange: (dateString: string) => void,
  error: string,
};

const EggEditorRenderer = ({
  mode,
  navigation,
  form,
  chickens,
  handlePickItem,
  onDateChange,
  toggleDamaged,
  onSaveForm,
  error,
}: Props) => (
  <Container>
    <Header>
      <Left />
      <Body>
        <Title>{`${mode} Egg`}</Title>
      </Body>
      <Right>
        <Button transparent onPress={() => navigation.goBack()}>
          <Text>Cancel</Text>
        </Button>
      </Right>
    </Header>
    <Content>
      <View padder>
        {error && <Text style={styles.error}>{error}</Text>}
        <FieldGroup
          control={form}
          render={({ invalid }) => (
            <Form>
              <Field
                control={form.get('date')}
                strict={false}
                render={({ value }) => (
                  <DatePicker
                    label="Date"
                    value={value}
                    onDateChange={onDateChange}
                    maximumDate={new Date(moment().valueOf())}
                  />
                )}
              />

              <Field
                control={form.get('chickenId')}
                render={({ value: chickenId, errors, touched }) => (
                  <View style={styles.pickerContainer}>
                    <Item inlineLabel error={touched && errors !== null}>
                      <Label style={styles.label}>Chicken:</Label>
                      <View style={{ flex: 1 }}>
                        <Picker
                          note
                          mode="dropdown"
                          textStyle={
                            chickenId !== '' ? styles.pickerText : null
                          }
                          selectedValue={chickenId || ''}
                          onValueChange={handlePickItem}
                        >
                          <Picker.Item label="Select a chicken" value="" />
                          <Picker.Item label="I'm not sure" value="unknown" />
                          {Object.keys(chickens || {}).map(key => (
                            <Picker.Item
                              key={key}
                              label={chickens[key].name}
                              value={key}
                            />
                          ))}
                        </Picker>
                      </View>
                      {touched && errors && <Icon name="alert" />}
                    </Item>
                    {errors
                      && touched && (
                        <Text style={styles.error}>
                          Select a chicken or &quot;I&apos;m not sure&quot;
                        </Text>
                    )}
                  </View>
                )}
              />

              <FieldControl
                name="weight"
                render={formProps => <FormInput {...formProps} />}
                meta={{
                  label: 'Weight (g)',
                  keyboardType: 'decimal-pad',
                }}
              />

              <FieldControl
                name="damaged"
                render={({ value }) => (
                  <View style={styles.checkboxContainer}>
                    <CheckBox
                      onPress={() => toggleDamaged(!value)}
                      checked={value}
                    />
                    <Text style={styles.checkboxLabel}>
                      Egg was damaged or broken
                    </Text>
                  </View>
                )}
              />

              <FieldControl
                name="notes"
                render={formProps => <FormInput {...formProps} />}
                meta={{
                  label: 'Notes',
                  multiline: true,
                  numberOfLines: 5,
                  maxLength: 250,
                }}
              />

              <Button
                block
                disabled={invalid}
                onPress={onSaveForm}
                style={styles.submit}
              >
                <Text>Save</Text>
              </Button>
            </Form>
          )}
        />
      </View>
    </Content>
  </Container>
);

export default EggEditorRenderer;
