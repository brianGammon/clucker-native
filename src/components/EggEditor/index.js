import EggEditor from './EggEditor';

export default EggEditor;
