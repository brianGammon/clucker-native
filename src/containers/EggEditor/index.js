import React from 'react';
import { Text, View, Button } from 'react-native';

export interface Props {
  navigation: any;
}

export default class EggEditor extends React.Component<Props> {
  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ fontSize: 30 }}>Add or Edit an Egg Here</Text>
        <Button onPress={() => navigation.goBack()} title="Dismiss" />
      </View>
    );
  }
}
