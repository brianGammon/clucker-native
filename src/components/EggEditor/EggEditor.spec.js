import React from 'react';
import configureStore from 'redux-mock-store';
import EggEditor from '.';
import mockNavigation from '../../../jest/mockNavigation';
import sampleData from '../../../jest/sampleData';
import '../../../jest/mockDateHelper';

describe('EggEditor component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);

  test('Should render connected EggEditor with default chicken, egg, and date', () => {
    const paramReturned = {
      getParam: {
        chickenId: 'chicken1',
        date: '2018-02-10',
        eggId: 'egg1',
      },
    };
    const navigation = mockNavigation(paramReturned);
    const wrapper = shallow(
      <EggEditor store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render connected EggEditor with no defaults', () => {
    const paramReturned = {};
    const navigation = mockNavigation(paramReturned);
    const wrapper = shallow(
      <EggEditor store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
