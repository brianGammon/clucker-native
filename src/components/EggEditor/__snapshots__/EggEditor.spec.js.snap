// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EggEditor component: Should render connected EggEditor with default chicken, egg, and date 1`] = `
<EggEditorRenderer
  chickens={
    Object {
      "chicken1": Object {
        "breed": "Cuckoo Maran",
        "hatched": "2016-10-05",
        "name": "Chicken1",
        "photoPath": "uploads/user:user1/flock:flock1/1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=689dfae8-c034-46ae-9984-db5c4ee3fe72",
      },
      "chicken2": Object {
        "breed": "Black Australorpe",
        "hatched": "2016-10-05",
        "name": "Chicken2",
        "photoPath": "uploads/user:user1/flock:flock1/8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg?alt=media&token=0ae6a73d-7027-4d18-8ab2-3f40340496c5",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg?alt=media&token=0e0d4dea-5104-4cfe-9a49-f87d4b412e24",
      },
      "chicken3": Object {
        "breed": "Welsummer",
        "hatched": "2016-10-05",
        "name": "Chicken3",
        "photoPath": "uploads/user:user1/flock:flock1/1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg?alt=media&token=54a9eb90-7a2a-4370-a030-7cb3fa3bb738",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg?alt=media&token=708ffb11-4a0a-4dab-9633-27a2c0c74d18",
      },
      "chicken4": Object {
        "breed": "Welsummer",
        "hatched": "2016-10-05",
        "name": "Chicken4",
        "photoPath": "uploads/user:user1/flock:flock1/9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg?alt=media&token=77141d6f-bb85-4174-817b-ad2cad174315",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg?alt=media&token=baa3c541-626a-45e7-b75c-1eab00454476",
      },
    }
  }
  error=""
  form={
    FormGroup {
      "_onDisabledChange": Array [],
      "_pendingChange": false,
      "_pendingDirty": false,
      "_pendingTouched": false,
      "asyncValidator": null,
      "controls": Object {
        "chickenId": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "chicken1",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": [Function],
          },
          "value": "chicken1",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "chickenName": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "Chicken1",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": "Chicken1",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "damaged": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": true,
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": false,
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": true,
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "date": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "2018-02-06",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "2018-02-06",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "notes": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "Chicken1’s first egg in 3 months!! ",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": "Chicken1’s first egg in 3 months!! ",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "weight": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "65.9",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "65.9",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
      },
      "errors": null,
      "get": [Function],
      "getError": [Function],
      "handleSubmit": [Function],
      "hasError": [Function],
      "meta": Object {},
      "patchValue": [Function],
      "pristine": true,
      "reset": [Function],
      "setValue": [Function],
      "stateChanges": Observable {
        "observers": Array [],
      },
      "status": "VALID",
      "statusChanges": Observable {
        "observers": Array [],
      },
      "submitted": false,
      "touched": false,
      "validator": null,
      "validatorOrOpts": Object {
        "asyncValidators": null,
        "updateOn": null,
        "validators": null,
      },
      "value": Object {
        "chickenId": "chicken1",
        "chickenName": "Chicken1",
        "damaged": true,
        "date": "2018-02-06",
        "notes": "Chicken1’s first egg in 3 months!! ",
        "weight": "65.9",
      },
      "valueChanges": Observable {
        "observers": Array [],
      },
    }
  }
  handlePickItem={[Function]}
  mode="Edit"
  navigation={
    Object {
      "getParam": [MockFunction] {
        "calls": Array [
          Array [
            "chickenId",
            null,
          ],
          Array [
            "eggId",
            null,
          ],
          Array [
            "date",
            null,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": "chicken1",
          },
          Object {
            "isThrow": false,
            "value": "-L4geyAR0BkMrAjiF7NZ",
          },
          Object {
            "isThrow": false,
            "value": "2018-02-10",
          },
        ],
      },
    }
  }
  onDateChange={[Function]}
  onSaveForm={[Function]}
  toggleDamaged={[Function]}
/>
`;

exports[`EggEditor component: Should render connected EggEditor with no defaults 1`] = `
<EggEditorRenderer
  chickens={
    Object {
      "chicken1": Object {
        "breed": "Cuckoo Maran",
        "hatched": "2016-10-05",
        "name": "Chicken1",
        "photoPath": "uploads/user:user1/flock:flock1/1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=689dfae8-c034-46ae-9984-db5c4ee3fe72",
      },
      "chicken2": Object {
        "breed": "Black Australorpe",
        "hatched": "2016-10-05",
        "name": "Chicken2",
        "photoPath": "uploads/user:user1/flock:flock1/8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg?alt=media&token=0ae6a73d-7027-4d18-8ab2-3f40340496c5",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_8BF48E4E-F5EF-4DCC-91C1-746CF33586FB.jpeg?alt=media&token=0e0d4dea-5104-4cfe-9a49-f87d4b412e24",
      },
      "chicken3": Object {
        "breed": "Welsummer",
        "hatched": "2016-10-05",
        "name": "Chicken3",
        "photoPath": "uploads/user:user1/flock:flock1/1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg?alt=media&token=54a9eb90-7a2a-4370-a030-7cb3fa3bb738",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_1A2F5FA5-FE29-4D43-9E4A-E44F19CE44B9.jpeg?alt=media&token=708ffb11-4a0a-4dab-9633-27a2c0c74d18",
      },
      "chicken4": Object {
        "breed": "Welsummer",
        "hatched": "2016-10-05",
        "name": "Chicken4",
        "photoPath": "uploads/user:user1/flock:flock1/9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg",
        "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg?alt=media&token=77141d6f-bb85-4174-817b-ad2cad174315",
        "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg",
        "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_9B6AF4A8-B948-4BF1-AEB6-9D78AD9A382E.jpeg?alt=media&token=baa3c541-626a-45e7-b75c-1eab00454476",
      },
    }
  }
  error=""
  form={
    FormGroup {
      "_onDisabledChange": Array [],
      "_pendingChange": false,
      "_pendingDirty": false,
      "_pendingTouched": false,
      "asyncValidator": null,
      "controls": Object {
        "chickenId": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": Object {
            "required": true,
          },
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "INVALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": [Function],
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "chickenName": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "damaged": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": false,
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": false,
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": false,
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "date": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "2018-05-15",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "2018-05-15",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "notes": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
        "weight": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": null,
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "VALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
      },
      "errors": null,
      "get": [Function],
      "getError": [Function],
      "handleSubmit": [Function],
      "hasError": [Function],
      "meta": Object {},
      "patchValue": [Function],
      "pristine": true,
      "reset": [Function],
      "setValue": [Function],
      "stateChanges": Observable {
        "observers": Array [],
      },
      "status": "INVALID",
      "statusChanges": Observable {
        "observers": Array [],
      },
      "submitted": false,
      "touched": false,
      "validator": null,
      "validatorOrOpts": Object {
        "asyncValidators": null,
        "updateOn": null,
        "validators": null,
      },
      "value": Object {
        "chickenId": "",
        "chickenName": "",
        "damaged": false,
        "date": "2018-05-15",
        "notes": "",
        "weight": "",
      },
      "valueChanges": Observable {
        "observers": Array [],
      },
    }
  }
  handlePickItem={[Function]}
  mode="Add"
  navigation={
    Object {
      "getParam": [MockFunction] {
        "calls": Array [
          Array [
            "chickenId",
            null,
          ],
          Array [
            "eggId",
            null,
          ],
          Array [
            "date",
            null,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": null,
          },
          Object {
            "isThrow": false,
            "value": null,
          },
          Object {
            "isThrow": false,
            "value": null,
          },
        ],
      },
    }
  }
  onDateChange={[Function]}
  onSaveForm={[Function]}
  toggleDamaged={[Function]}
/>
`;
